lib/ProductOpener/Users.pm

```diff
@@ -615,19 +621,108 @@ sub check_edit_owner($user_ref, $errors_ref) {
 }
 
 
-sub init_user() {
 
-	my $user_id = undef ;
-	my $user_ref = undef;
-	my $org_ref = undef;
+=head2 open_user_session($user_ref, $request_ref)
+
+Open a session, store it in the user object, and return a cookie with the session id in the request object.
+
+=head3 Arguments
+
+=head4 User object $user_ref
+
+=head4 Request object $request_ref
+
+=head3 Return values
+
+The cookie is returned in $request_ref
+
+=cut
+
+sub open_user_session($user_ref, $request_ref) {
+
+	my $user_id = $user_ref->{'userid'};
+
+	# Maximum of sessions for a given user
+	my $max_session = 10 ;
+
+	# Generate a secure session key, store the cookie
+	my $user_session = generate_token(64);
+	$log->context->{user_session} = $user_session;
+
+	defined $user_ref->{'user_sessions'} or $user_ref->{'user_sessions'} = {};
+
+	# Check if we need to delete the oldest session
+	# delete $user_ref->{'user_session'};
+	if ((scalar keys %{$user_ref->{'user_sessions'}}) >= $max_session) {
+		my %user_session_stored = %{$user_ref->{'user_sessions'}} ;
+
+		# Find the older session and remove it
+		my @session_by_time = sort { $user_session_stored{$a}{'time'} <=>
+						$user_session_stored{$b}{'time'} } (keys %user_session_stored);
+
+		while (($#session_by_time + 1)> $max_session)
+		{
+			my $oldest_session = shift @session_by_time;
+			delete $user_ref->{'user_sessions'}{$oldest_session};
+		}
+	}
```

Comment:
I think it would be better to have that kind of stuff in a separate sub, for clarity when reading, because it puts the reader out of the flow, while it's easy to isolate (and would ease testing).